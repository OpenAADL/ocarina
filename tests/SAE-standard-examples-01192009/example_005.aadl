-- test2 4.5 Subcomponents draft-01192009

package Redundancy
public
abstract Triple
features
input: in data port;
output: out data port;
end Triple;
abstract implementation Triple.impl
subcomponents
MyProcess: abstract Calculate [3];
MyVoter: abstract Voter;
connections
extinput: port input -> MyProcess.input
{ Connection_Pattern => One_To_Many; };
tovoter: port MyProcess.output -> MyVoter.input
{ Connection_Pattern => One_To_One; };
extoutput: port MyVoter.output -> output;
end Triple.impl;
abstract Calculate
features
input: in data port;
output: out data port;
end Calculate;
abstract Voter
features

input: in data port [3];
-- replace by this for test no crash
-- because array in features not supported
-- input: in data port;

output: out data port;
end Voter;
end Redundancy;