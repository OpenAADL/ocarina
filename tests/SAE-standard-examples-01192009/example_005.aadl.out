example_005.aadl:16:22:  (property term) does not point to anything or to something unreachable
Cannot analyze AADL specifications
